Do not install Selenium in "Build:" jobs

TanayParikh opened the pull request:
Failing build: https://dev.azure.com/dnceng/internal/_build/results?buildId=1014505

Doug pointed out how Selenium was failing to install in official builds. Turns out the E2ETest are building when they shouldn't be (thanks @BrennanConroy!).

TanayParikh committed:
Mark E2ETest as test project

HaoK reviewed (approved):
Ah right because we don't follow the naming convention

dougbu reviewed (approved):
Please push this branch to the internal (AzDO) dotnet-aspnetcore and starting an aspnetcore-ci-official build manually (a PR would build tests and we don't want that). Then download the binary log to confirm nothing installs Selenium. We may have other test projects w/ unconventional names and no override. If that works, :shipit:

TanayParikh committed:
Use IsIntegrationTestProject

TanayParikh committed:
Add Condition="'$(ExcludeFromBuild)' != 'true'"

dougbu reviewed (commented):
May need to change src/ProjectTemplates/test/ProjectTemplates.Tests.csproj as well

TanayParikh committed:
Update ProjectTemplates.Tests.csproj

dougbu committed:
`$(ExcludeFromBuild)` -> `$(SkipTestBuild)`
- `$(ExcludeFromBuild)` isn't set until Directory.Build.targets is read
- `$(SkipTestBuild)` is a global property (if set)

dougbu commented:
I started https://dev.azure.com/dnceng/internal/_build/results?buildId=1024838 but will cancel it because public build tests fix in "Build: Windows x64/x86" job.

dougbu commented:
added `auto-merge` label because "Build:" jobs are done and they didn't install Selenium 🎉

dougbu commented:
Ah well, `auto-merge` doesn't work when something fails in the quarantined pipeline ☹️
